src/test/java/edu/hm/hafner/analysis/ModuleDetectorTest.java

```diff
@@ -124,14 +129,203 @@ void shouldIdentifyModuleByReadingMavenPomWithoutName() {
+    }
+
+    @Test
+    void shouldEnsureThatGradleSettingsCanParseFormat1() throws IOException {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.SETTINGS_GRADLE,
+            });
+            when(stub.open(anyString())).thenAnswer(fileName -> read("settings-1.gradle"));
+        });
+
+        String gradleWorkspace = PREFIX + PATH_PREFIX_GRADLE;
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+        assertThat(detector.guessModuleName(gradleWorkspace + "build/reports/something.txt"))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_ROOT);
+    }
+
+    @Test
+    void shouldEnsureThatGradleSettingsCanParseFormat2() throws IOException {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.SETTINGS_GRADLE,
+            });
+            when(stub.open(anyString())).thenAnswer(fileName -> read("settings-2.gradle"));
+        });
+
+        String gradleWorkspace = PREFIX + PATH_PREFIX_GRADLE;
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+        assertThat(detector.guessModuleName(gradleWorkspace + "build/reports/something.txt"))
+                .isEqualTo("root-project-2");
+    }
+
+    @Test
+    void shouldEnsureThatGradleSettingsCanParseFormat3() throws IOException {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.SETTINGS_GRADLE,
+            });
+            when(stub.open(anyString())).thenAnswer(fileName -> read("settings-3.gradle"));
+        });
+
+        String gradleWorkspace = PREFIX + PATH_PREFIX_GRADLE;
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+        assertThat(detector.guessModuleName(gradleWorkspace + "build/reports/something.txt"))
+                .isEqualTo("root-project-3");
+    }
+
+    @Test
+    void shouldEnsureThatGradleSettingsCanParseFormat4() throws IOException {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.SETTINGS_GRADLE,
+            });
+            when(stub.open(anyString())).thenAnswer(fileName -> read("settings-4.gradle"));
+        });
+
+        String gradleWorkspace = PREFIX + PATH_PREFIX_GRADLE;
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+        assertThat(detector.guessModuleName(gradleWorkspace + "build/reports/something.txt"))
+                .isEqualTo("root-project-4");
+    }
+
+    @Test
+    void shouldEnsureThatGradleSettingsWithoutProjectNameIgnored() throws IOException {
```

Comment:
Added another test with a settings.gradle file that does not rewrite rootProject.name, to ensure that it falls back to the build.gradle base-directory detection.